sources/packages/api/src/msfaa-integration/msfaa-integration.service.ts

```diff
@@ -0,0 +1,190 @@
+   */
+  createMSFAARequestContent(
+    msfaaRecords: MSFAARecord[],
+    fileSequence: number,
+    totalSINHash: number,
+  ): MSFAARequestFileLine[] {
+    const processDate = new Date();
+    const msfaaFileLines: MSFAARequestFileLine[] = [];
+    // Header record
+    const msfaaHeader = new MSFAAFileHeader();
+    msfaaHeader.transactionCode = TransactionCodes.MSFAARequestHeader;
+    msfaaHeader.processDate = processDate;
+    msfaaHeader.provinceCode = this.msfaaConfig.provinceCode;
+    msfaaHeader.sequence = fileSequence;
+    msfaaFileLines.push(msfaaHeader);
+    // Detail records
+    const fileRecords = msfaaRecords.map((msfaaRecord) => {
+      const msfaaDetail = new MSFAAFileDetail();
+      msfaaDetail.transactionCode = TransactionCodes.MSFAARequestDetail;
+      msfaaHeader.processDate = processDate;
+      msfaaDetail.msfaaNumber = msfaaRecord.msfaaNumber;
+      msfaaDetail.sin = msfaaRecord.sin;
+      msfaaDetail.institutionCode = msfaaRecord.institutionCode;
+      msfaaDetail.birthDate = msfaaRecord.birthDate;
+      msfaaDetail.surname = msfaaRecord.surname;
+      msfaaDetail.givenName = msfaaRecord.givenName;
+      msfaaDetail.genderCode = getGenderCode(msfaaRecord.gender);
+      msfaaDetail.maritalStatusCode = getMaritalStatusCode(
+        msfaaRecord.maritalStatus,
+      );
+      msfaaDetail.addressLine1 = msfaaRecord.addressLine1;
+      msfaaDetail.addressLine2 = msfaaRecord.addressLine2
+        ? msfaaRecord.addressLine2
+        : "";
+      msfaaDetail.city = msfaaRecord.city;
+      msfaaDetail.province = msfaaRecord.province;
+      msfaaDetail.postalCode = msfaaRecord.postalCode;
+      msfaaDetail.country = msfaaRecord.country;
+      msfaaDetail.phone = msfaaRecord.phone;
+      msfaaDetail.email = msfaaRecord.email;
+      msfaaDetail.offeringIntensityCode = getOfferingIntensityCode(
+        msfaaRecord.offeringIntensity,
+      );
+      return msfaaDetail;
+    });
+    msfaaFileLines.push(...fileRecords);
+    // Footer or Trailer record
+    const msfaaFooter = new MSFAAFileFooter();
+    msfaaFooter.transactionCode = TransactionCodes.MSFAARequestTrailer;
+    msfaaFooter.totalSINHash = totalSINHash;
+    msfaaFooter.recordCount = msfaaRecords.length;
+    msfaaFileLines.push(msfaaFooter);
+
+    return msfaaFileLines;
+  }
+
+  /**
+   * Converts the MSFAAFileLines to the final content and upload it.
+   * @param MSFAAFileLines Array of lines to be converted to a formatted fixed size file.
+   * @param remoteFilePath Remote location to upload the file (path + file name).
+   * @returns Upload result.
+   */
+  async uploadContent(
+    msfaaFileLines: MSFAARequestFileLine[],
+    remoteFilePath: string,
+  ): Promise<MSFAAUploadResult> {
+    // Generate fixed formatted file.
+    const fixedFormattedLines: string[] = msfaaFileLines.map(
+      (line: MSFAARequestFileLine) => line.getFixedFormat(),
+    );
+    const msfaaFileContent = fixedFormattedLines.join("\r\n");
```

Comment:
a general qn, if the server is Linux, will it ignore `carriage return` or consider it as `line feed`(i.e ` \n\n`)?

Reply:
I am not sure, but I think it will not be ignored and in our case, a line is fed

Reply:
For the CRA migration, the break lines were defined as _"\r\n"_ and the files were properly received by the CRA. We follow the same that was being done on PROD files.
@abschwenker can confirm on PROD files the expectation for MSFAA also.

Reply:
correct, these files use the same mechanisms to be sent. So the way the CRA files are sent can be mirrored

Reply:
https://superuser.com/questions/374028/how-are-n-and-r-handled-differently-on-linux-and-windows, If it's tested in both servers, then, I am good

Reply:
@ann-aot, they don't need to be tested on both servers, they will not be used by two different servers.
The files have one way to be generated and we are mirroring the PROD files. These files are going to be passed to Windows servers but the final destination are the federal servers that, we have no idea what they are, we do not need to know, we just need to know how the PROD files looks like and follow it 😉 
Hope that it is clear.